lucene/core/src/java/org/apache/lucene/index/SegmentCommitInfo.java

```diff
@@ -91,16 +91,25 @@
    * @param docValuesGen
    *          DocValues generation number (used to name doc-values updates files)
    */
-  public SegmentCommitInfo(SegmentInfo info, int delCount, int softDelCount, long delGen, long fieldInfosGen, long docValuesGen) {
+  public SegmentCommitInfo(SegmentInfo info, int delCount, int softDelCount,
```

Comment:
I think we should use the  `setNext*Gen` setters instead of this. We need them anyway and then we have only one way to set these values. It might also make sense to assert that the next*Gen is > than the actual gen in these setters.